plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -209,9 +232,25 @@ def get_observations(
                         f"More data is available on the API (num_found={num_found}) - setting has_more_pages=True..."
                     )
                     has_more_pages = True
+
+            if not observations:
```

Comment:
Can we add a comment to why we've done this? I know for myself this time tomorrow I won't understand this is to catch dropping any duped observations and not wanting to move the offset